auctions/dutch-auction/src/main.sw

```diff
@@ -0,0 +1,219 @@
+contract;
+
+dep abi;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    address::Address,
+    assert::require,
+    block::height,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::msg_asset_id, msg_amount},
+    contract_id::ContractId,
+    identity::Identity,
+    logging::log,
+    result::*,
+    revert::revert,
+    storage::StorageMap,
+    token::{force_transfer_to_contract, transfer_to_output}
+};
+
+use abi::DutchAuction;
+use data_structures::Auction;
+use errors::{AuthorizationError, BidError, SetupError, TechnicalError, TimeError};
+use events::{AuctionCancelledEvent, CreatedAuctionEvent, WinningBidEvent};
+
+storage {
+    /// The Admin Address
+    admin: Identity,
+    /// Mapping an auction_id to its respective auction, allowing for multiple auctions to happen simultaneously
+    auctions: StorageMap<u64,
+    Auction>, /// Tracking how many auctions have been made till now
+    auction_count: u64,
+    /// Whether the constructor has been called
+    initialized: bool,
+}
+
+impl DutchAuction for Contract {
+    /// This function should be called right after being deployed, so the admin can be set
+    fn constructor(admin: Identity) {
+        require(!storage.initialized, AuthorizationError::CannotReinitialize);
+        storage.admin = admin;
+        storage.initialized = true;
+    }
+
+    /// Returns the current price for the auction corresponding to the auction_id
+    fn price(auction_id: u64) -> u64 {
+        validate_id(auction_id);
+
+        calculate_price(auction_id)
+    }
+
+    /// Bids in the given auction_id, is a win if the amount and asset are correct
+    fn bid(auction_id: u64) {
+        // In a Dutch auction the first bid wins
+        require(storage.initialized, TechnicalError::ContractNotYetInitialized);
```

Comment:
Remove require. Your id validation checks to see if the ID is in the valid range (i.e. also not 0)